src/codegen/codegen_ispc_visitor.cpp

```diff
@@ -292,6 +305,31 @@ std::string CodegenIspcVisitor::ptr_type_qualifier() {
     }
 }
 
+std::string CodegenIspcVisitor::global_var_structure_type_decorator() {
+    if (wrapper_codegen) {
+        return CodegenCVisitor::global_var_structure_type_decorator();
+    } else {
+        return "uniform ";  // @note: extra space needed to separate qualifier from var name.
+    }
+}
```

Comment:
I would like to discuss this bit in detail - I am thinking isn't it would be simple if somehow we avoid execution of ISPC codegen completely and use C++ visit? Lets discuss this offline.

Reply:
probably it is possible. However, the ispc file would be completly empty. Does it work in that case?

Reply:
> Does it work in that case?

I hope! need to test...

Reply:
if I do not write anything in the ispc file I get:
`Error: Premature end of file: syntax error, unexpected end of file.`

Reply:
ok, solved